### PR TITLE
[ENH] Add variance_factor to DummyProbaRegressor

### LINKED ISSUE #7: (wish)list of probabilistic regressors to implement or to interface
A wishlist for probabilistic regression methods to implement or interface.
This is copied from the R counterpart https://github.com/mlr-org/mlr3proba/issues/32 .
Number of stars at the end is estimated difficulty or time investment.

GLM
- [ ] generalized linear model(s) with regression link, e.g., Gaussian *
- [ ] generalized linear model(s) with count link, e.g., Poisson *
- [ ] heteroscedastic linear regression ***
- [ ] Bayesian GLM where conjugate priors are available, e.g., GLM with Gaussian link ***

KRR aka Gaussian process regression
- [ ] vanilla kernel ridge regression with fixed kernel parameters and variance *
- [ ] kernel ridge regression with MLE for kernel parameters and regularization parameter **
- [ ] heteroscedastic KRR or Gaussian processes ***

CDE
- [ ] variants of conditional density estimation (Nadaraya-Watson type) **
- [ ] reduction to density estimation by binning of input variables, then apply unconditional density estimation **

Tree-based
- [ ] probabilistic regression trees **

Neural networks
- [ ] interface tensorflow probability - some hard-coded NN architectures **
- [ ] generic tensorflow probability interface - some hard-coded NN architectures ***

Bayesian toolboxes
- [ ] generic pymc3 interface ***
- [ ] generic pyro interface ****
- [ ] generic Stan interface ****
- [ ] generic JAGS interface ****
- [ ] generic BUGS interface ****
- [ ] generic Bayesian interface - prior-valued hyperparameters *****

Pipeline elements for target transformation
- [ ] distr fixed target transformation **
- [ ] distr predictive target calibration **

Composite techniques, reduction to deterministic regression
- [ ] stick mean, sd, from a deterministic regressor which already has these as return types into some location/scale distr family (Gaussian, Laplace) *
- [ ] use model 1 for the mean, model 2 fit to residuals (squared, absolute, or log), put this in some location/scale distr family (Gaussian, Laplace) **
- [ ] upper/lower thresholder for a regression prediction, to use as a pipeline element for a forced lower variance bound **
- [ ] generic parameter prediction by elicitation, output being plugged into parameters of a distr object not necessarily scale/location ****
- [ ] reduction via bootstrapped sampling of a determinstic regressor **

Ensembling type pipeline elements and compositors
- [ ] simple bagging, averaging of pdf/cdf **
- [ ] probabilistic boosting ***
- [ ] probabilistic stacking ***

baselines
- [ ] always predict a Gaussian with mean = training mean, var = training var *
- [ ] IMPORTANT as featureless baseline: reduction to distr/density estimation to produce an unconditional probabilistic regressor **
- [ ] IMPORTANT as deterministic style baseline: reduction to deterministic regression, mean = prediction by det.regressor, var = training sample var, distr type = Gaussian (or Laplace) **

Other reduction from/to probabilistic regression
- [ ] reducing deterministic regression to probabilistic regression - take mean, median or mode **
- [ ] reduction(s) to quantile regression, use predictive quantiles to make a distr ***
- [ ] reducing deterministic (quantile) regression to probabilistic regression - take quantile(s) **
- [ ] reducing interval regression to probabilistic regression - take mean/sd, or take quantile(s) **
- [ ] reduction to survival, as the sub-case of no censoring **
- [ ] reduction to classification, by binning ***

### PR DESCRIPTION
**Pull request by hrshx3o5o6**
This PR adds a variance_factor option to the `DummyProbaRegressor` when the underlying strategy is normal.

Previously, the normal strategy correctly predicted a Normal distribution with the empirical standard deviation. However, probabilistic modeling often requires structural baselines where the prediction uncertainty must be explicitly hedged or aggressively reduced. This enhancement allows practitioners to natively scale the standard deviation to predict structural baselines where the variance is equal to $\sigma_{training}^2 \times \text{factor}$. I ran validations locallyt and it both passed. 

Resolves #7 , implementing a still unchecked baseline titled "always predict a Gaussian with mean = training mean, var = training var".

### CODE REVIEW

**Comment by hrshx3o5o6** [comment]
> Thanks. Can you give an example in which this would be useful?

The main use case is creating tunable, conservative probabilistic baselines. In real-world applications (finance, supply chain, forecasting), future data is often more volatile than training data. The standard DummyProbaRegressor can be unrealistically confident.

By setting variance_factor > 1, users can easily construct a baseline with wider predictoin levels. This provides a much stricter benchmark to beat for uncertainty metrics like CRPS or Log-Likelihood than a naive, unscaled dummy model.

**Comment by fkiraly** [comment]
How would you set the `variance_factor` in practice?

**Comment by hrshx3o5o6** [comment]
> How would you set the `variance_factor` in practice?

In practice, the value of the variance_factor could be determined in these ways:

- Hyperparameter tuning: suppose we are stacking models or using a fallback regressor; you can use GridSearchCV to find an optimal variance factor.

- Domain knowledge / heuristics: In specific domains like finance, for example, practitioners might have a volatility multiplier in mind (e.g., "sales might be volatile by 20% this quarter"). Hence, ensuring that domain knowledge is injected natively into the baseline.

- Calibration from residual statistics: lets say standard dummy data only covers 80% of the test set but the target requires 95% percent coverage, this can be taken care of by manually tuning up the `variance_factor` until the baseline achieves emperical coverage.

**Comment by hrshx3o5o6** [comment]
@fkiraly just checking in to see if there is any update on this?
